content/articles/sentiment-analysis-with-naive-bayes-algorithms/index.md

```diff
@@ -0,0 +1,1035 @@
+        text = text.replace(char, "")
+ 
+ # lemmatize the words and join back into string text
+    text = " ".join([wordnet_lemmatizer.lemmatize(word) for word in word_tokenize(text)])
+ 
+ return text
+
+
+
+class DenseTransformer(TransformerMixin):
+ def fit(self, x, y=None, **fit_params):
+ return self
+
+ @staticmethod
+ def transform(x, y=None, **fit_params):
+ return x.todense()
+ 
+ def __str__(self):
+ return "DenseTransformer()"
+ 
+ def __repr__(self):
+ return self.__str__()
+ 
+ 
+class CleanTextTransformer(TransformerMixin):
+ def fit(self, x, y=None, **fit_params):
+ return self
+ 
+ @staticmethod
+ def transform(x, y=None, **fit_params):
+ return np.vectorize(clean_text)(x)
+
+ def __str__(self):
+ return 'CleanTextTransformer()'
+
+ def __repr__(self):
+ return self.__str__()
+
+
+
+def load_imdb_sentiment_analysis_dataset(imdb_data_path, seed=123):
+ """This loads the IMDB movie review dataset.
+    # arguments
+        imdb_data_path: string.
+        seed: int.
+    # returns
+        Training data and validation data, must be tuple.
+        Training data has 25000 samples
+        Test data has 25000 samples
+        Categories: 2 (0 - negative, 1 - positive)
+    # references
+        Bmass et al., http://www.aclweb.org/anthology/P11-1015
+        Download and uncompress archive from:
+        http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz
+    """
+
+ # Load training data:
+
+    train_texts = []
+    train_labels = []
+ for category in ['pos', 'neg']:
+ print(f"loading train: {category} ...")
+        train_path = os.path.join(imdb_data_path, 'train', category)
+ for fname in tqdm(sorted(os.listdir(train_path))):
+ if fname.endswith('.txt'):
+ with open(os.path.join(train_path, fname), encoding="utf-8") as f:
+                    train_texts.append(f.read())
+                train_labels.append(0 if category == 'neg' else 1)
+
+ # Load validation data:
+
+    test_texts = []
+    test_labels = []
+ for category in ['pos', 'neg']:
+ print(f"loading test: {category} ...")
+        test_path = os.path.join(imdb_data_path, 'test', category)
+ for fname in tqdm(sorted(os.listdir(test_path))):
+ if fname.endswith('.txt'):
+ with open(os.path.join(test_path, fname), encoding="utf-8") as f:
+                    test_texts.append(f.read())
+                test_labels.append(0 if category == 'neg' else 1)
+
+ # Shuffle the data and labels of our training set.
+    random.seed(seed)
+    random.shuffle(train_texts)
+    random.seed(seed)
+    random.shuffle(train_labels)
+
+ return ((np.array(train_texts), np.array(train_labels)),
+            (np.array(test_texts), np.array(test_labels)))
+```
+
+Categorical and Gaussian naive Bayes models keep flagging errors because they can't take the samples of data provided all at once for some reason which might be due to memory allocation. Therefore, a separate function was written for the two to take in data samples in a particular size per batch.
```

Comment:
Identify the cause of the problem. Raising unanswered questions might confuse the reader.